### ISSUE TITLE
Component.get_context_data : add *args and **kwargs to match the signature when overridden

### ISSUE THREAD

**Issue by aymericderbois** [EmilStenstrom/django-components#187]
Hey,

For now the signature of the "get_context_data" method is the following:
```python
    def get_context_data(self):
        return {}
```
It could be nice to change signature of Component.get_context_data to add *args and **kwargs like this :
```python
    def get_context_data(self, *args, **kwargs):
        return {}
```

What do you think about it ?

Aymeric,

**Comment by EmilStenstrom** [comment]
Could you say a bit more about why your want that?

**Comment by aymericderbois** [comment]
Yes, indeed I should have given a more complete explanation on this request.

If I take the example of the "calendar" component from the documentation :
```python
@component.register("calendar")
class Calendar(component.Component):
    # Templates inside `[your apps]/components` dir and `[project root]/components` dir will be automatically found. To customize which template to use based on context
    # you can override def get_template_name() instead of specifying the below variable.
    template_name = "calendar/calendar.html"

    # This component takes one parameter, a date string to show in the template
    def get_context_data(self, date):
        return {
            "date": date,
        }

    class Media:
        css = "calendar/calendar.css"
        js = "calendar/calendar.js"
```

- Pycharm will display the following warning: "method signature does not match signature of overridden method"
- Pylint will display the following warning: https://pylint.pycqa.org/en/latest/user_guide/messages/warning/arguments-differ.html

And in a slightly different example, if I set the type of the date parameter like this :
```python
@component.register("calendar")
class Calendar(component.Component):
    # Templates inside `[your apps]/components` dir and `[project root]/components` dir will be automatically found. To customize which template to use based on context
    # you can override def get_template_name() instead of specifying the below variable.
    template_name = "calendar/calendar.html"

    # This component takes one parameter, a date string to show in the template
    def get_context_data(self, date: str):
        return {
            "date": date,
        }

    class Media:
        css = "calendar/calendar.css"
        js = "calendar/calendar.js"
```
Then mypy will raise the following error: 
```
hollved/game/components/calendar/calendar.py:8: error: Signature of "get_context_data" incompatible with supertype "Component"
myapp/components/calendar/calendar.py:8: note:      Superclass:
myapp/components/calendar/calendar.py:8: note:          def get_context_data(self) -> Any
myapp/components/calendar/calendar.py:8: note:      Subclass:
myapp/components/calendar/calendar.py:8: note:          def get_context_data(self, date: str) -> Any
```

**Comment by EmilStenstrom** [comment]
Thank you for you explanation! I fixed this as part of the 0.23 release. Let me know how it works.
